python-frontend/src/main/java/org/sonar/python/semantic/v2/FunctionTypeBuilder.java

```diff
@@ -37,7 +37,7 @@
 
 import static org.sonar.python.tree.TreeUtils.locationInFile;
 
-public class FunctionTypeBuilder {
+public class FunctionTypeBuilder implements TypeBuilder {
```

Comment:
I still think we can simply use the [record builder library](https://github.com/Randgalt/record-builder) which will generate builders for us cause it does completely the same

Reply:
As discussed, I think we can introduce this later if we actually stick with records.
Even then, I'm not sure how important this is for us given I don't expect we will create that many more records.